doc/user-guide/push-notifications/Push-notifications-client-side.md

```diff
@@ -0,0 +1,158 @@
+## Using push notifications on client side
+
+There are just a few things the XMPP client application needs to receive the push notifications.
+One picture is worth a thousand words, hence, here you have two diagrams to follow:
+
+#### PubSub-full:
+ ![](pubsub_full_push_notifications_example_message_flow.png)
+
+#### PubSub-less:
+ ![](pubsub_less_push_notifications_example_message_flow.png)
+
+
+### Registering with a Push Service provider
+
+First, the client application has to get a device-specific token from the Push Service Provider
+(FCM or APNS). This process is different, depending on the platform, so please consult your Push
+Service Provider's manual to see how to get this token. For example,
+[here](https://firebase.google.com/docs/cloud-messaging/android/client) you can learn about setting
+up _FCM_ on _Android_ platform and
+[here](https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/HandlingRemoteNotifications.html#/apple_ref/doc/uid/TP40008194-CH6-SW1)
+you can learn about setting up _APNS_ on _iOS_ platform.
+
+After this step, your application shall be able to receive _FCM_ or _APNS_ token - it will be
+required in the next step of this tutorial.
+
+### Setting up an XMPP `pubsub` node
+
+This step is specific to the pubsub-full push configuration that you chose for your MongooseIM
+server. If you're running a pubsub-less configuration, skip to [this point](#enabling-push-notifications).
+
+#### Creating a new push node
+In this example `mypubsub.com` is a domain of the MongooseIM server that has [mod_pubsub][] enabled
+with the `push` node support.  The client sends the following stanza to the server:
+
+```xml
+<iq type='set'
+    to='pubsub.mypubsub.com'
+    id='create1'>
+  <pubsub xmlns='http://jabber.org/protocol/pubsub'>
+    <create node='punsub_node_for_my_private_iphone' type='push'/>
```

Comment:
This is just a bad pun. I know it would work (and is probably not a real pun) and it was used throughout this file but I just noticed it. `punsub`, really?

Reply:
It was there already, I just copied it as it was, also got confused by that "`punsub`", and then realised it is all consistently written everywhere, so 🤷‍♂ 😂

Reply:
It was there already, I just copied it as it was, also got confused by that "`punsub`", and then realised it is all consistently written everywhere, so 🤷‍♂ 😂